tests/unit/htdocs/class/cache/XoopsCacheEngineTest.php

```
@@ -0,0 +1,140 @@
+#[CoversClass(XoopsCacheEngine::class)]
+class XoopsCacheEngineTest extends TestCase
+{
+    private XoopsCacheEngine $engine;
+
+    protected function setUp(): void
+    {
+        require_once XOOPS_ROOT_PATH . '/class/cache/xoopscache.php';
+
+        // Create a concrete subclass for testing the abstract class
+        $this->engine = new class extends XoopsCacheEngine {
+            public function write($key, $value, $duration = null) { return true; }
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$value'. (undefined)

(UnusedFormalParameter)